api/helpers/gameStates/deal-cards.js

```diff
@@ -0,0 +1,73 @@
+const GameStatus = require('../../../utils/GameStatus.json');
+const DeckIds = require('../../../utils/DeckIds.json');
+const MoveType = require('../../../utils/MoveType.json');
+const GamePhase = require('../../../utils/GamePhase.json');
+
+module.exports = {
+  friendlyName: 'Deal Cards',
+
+  description: 'Creates initial GameStateRow for specified game, with cards dealt ot both players (5 for p0 and 6 for p1). Errors if Game has previous game or incorrec status.',
+
+  inputs: {
+    game: {
+      type: 'ref',
+      description: 'Game record to be initialized with first GameState',
+      required: true,
+    },
+  },
+
+  fn: async ({ game }, exits) => {
+    
+    try {
+      if (game.status !== GameStatus.CREATED) {
+        return exits.error({ message: 'Game has already started' });
+      }
+
+      const previousGameStates = await GameStateRow.find({ gameId: game.id});
+      if (previousGameStates?.length) {
```

Comment:
If we require/expect `game` to have populated its gameStates. But then what do we do if it's missing? Error? Or query for it?
```suggestion
      const previousGameStates = game.gameStates ?? await GameStateRow.find({ gameId: game.id});
      if (previousGameStates?.length) {
```

Reply:
Opted to throw because the intention is that it should always be an empty array when hitting this function and if not that indicates a problem